packages/alpha/src/types.ts

```diff
@@ -17,47 +17,64 @@ import {
 
 export * from '@cognite/sdk';
 
-export interface MonitoringTaskParametersCreate {
-  timeseriesExternalId: CogniteExternalId;
-  granularity?: string;
-  threshold: number;
+enum ModelExternalId {
+  Threshold = 'threshold',
+}
+
+export interface Model {
+  readonly externalId: ModelExternalId;
 }
 
-export interface MonitoringTaskParameters {
-  timeseriesExternalId: CogniteInternalId;
+export class MonitoringTaskThresholdModelCreate implements Model {
+  externalId: ModelExternalId = ModelExternalId.Threshold;
+  timeseriesExternalId: CogniteExternalId;
+  threshold: number;
+  granularity?: string;
+  constructor(
+    timeseriesExternalId: CogniteExternalId,
+    threshold: number,
+    granularity?: string
+  ) {
+    this.timeseriesExternalId = timeseriesExternalId;
+    this.granularity = granularity;
+    this.threshold = threshold;
+  }
+}
```

Comment:
we don't do classes here, this is overly complex for the SDK :)
```
export interface MonitoringTaskThresholdModelCreate {
  externalId: "threshold";
  timeseriesExternalId: CogniteExternalId;
  threshold: number;
  granularity?: string;
```
should be enough

Reply:
Tried this earlier. Interfaces don't seem to play well with literals, when constructing this and using it usually complains that (similar to) ` externalId: type "threshold" expected, got type string. `. Can mitigate this by not duck typing anymore and explicitly casting the object MonitoringTaskThresholdModelCreate (see new commit), which is not ideal either.

Reply:
Now users both have to declare explicitely which Model they're using in the cast and redundantly in the object iself with the externalId. By using the class and hidden enum this is not necessary.

Reply:
classes give different sorts of headaches, trust me on this one. js SDK has a long history of going from interfaces to classes and then back to interfaces.

If you wish to improve the developer experience you can define a type + const.
Example: 

```
export type ContextJobStatus = 'QUEUED' | 'RUNNING' | 'COMPLETED' | 'FAILED';

export const ContextJobStatus = {
  QUEUED: 'QUEUED' as ContextJobStatus,
  RUNNING: 'RUNNING' as ContextJobStatus,
  COMPLETED: 'COMPLETED' as ContextJobStatus,
  FAILED: 'FAILED' as ContextJobStatus,
};
```